ruby/command-t/lib/command-t/vim.rb

```diff
@@ -81,12 +81,12 @@ def wildignore_to_regexp(str)
           elsif pattern.match(%r{\A\*\.([^*]+)\z})
             # *.something (match file with extension at any level)
             '\.' + Regexp.escape($~[1]) + '\z'
+          elsif pattern.match(%r{\A\*/([^/]+)/\*\z})
+            # */something/* (match directories at any level)
+            '(\A|/)' + Regexp.escape($~[1]) + '/.*(/|\z)'
```

Comment:
Good call. Again, this highlights how difficult it is to translate something that's meant to match files and directories into a string matcher. 

The problem with the original `'(/|\z)'` at the end was that it would "incorrectly" catch e.g. `/dist{end of string}`. In most cases, the last part of the string is going to be "the file", so I made the assumption that we don't want to match on those strings. Which is why I added this part to the test: 

```ruby
      expect(regexp).to_not match('some/path/to/foo')
```